src/jit/compiler.h

```diff
@@ -661,23 +701,23 @@ class  LclVarDsc
     bool                IsFloatRegType() const
                         {
                             return
-#ifdef _TARGET_ARM_
+#ifdef FEATURE_HFA
                                 lvIsHfaRegArg ||
 #endif
                                 isFloatRegType(lvType);
                         }
-#ifdef _TARGET_ARM_
+#ifdef FEATURE_HFA
     var_types           GetHfaType() const
                         {
-                            assert(lvIsHfaRegArg);
-                            return lvIsHfaRegArg ? (lvHfaTypeIsFloat ? TYP_FLOAT : TYP_DOUBLE) : TYP_UNDEF;
+                            assert(lvIsHfa());
+                            return lvIsHfa() ? (lvHfaTypeIsFloat ? TYP_FLOAT : TYP_DOUBLE) : TYP_UNDEF;
```

Comment:
I think the assert is to make sure that we call this routine only on vars that are known to be hfa types.  Assert would make the detection of such a case easier, than TYP_UNDEF leading to some strange assert further along.

I would suggest to retain the assert, if the intent is that this routine is supposed to be called on those vars that are known to be a hfa.